go/apps/api/routes/v2_keys_update_credits/handler.go

```diff
@@ -0,0 +1,282 @@
+	"github.com/unkeyed/unkey/go/pkg/rbac"
+	"github.com/unkeyed/unkey/go/pkg/zen"
+)
+
+type Request = openapi.V2KeysUpdateCreditsRequestBody
+type Response = openapi.V2KeysUpdateCreditsResponse
+
+// Handler implements zen.Route interface for the v2 keys.updateCredits endpoint
+type Handler struct {
+	Logger      logging.Logger
+	DB          db.Database
+	Keys        keys.KeyService
+	Permissions permissions.PermissionService
+	Auditlogs   auditlogs.AuditLogService
+}
+
+// Method returns the HTTP method this route responds to
+func (h *Handler) Method() string {
+	return "POST"
+}
+
+// Path returns the URL path pattern this route matches
+func (h *Handler) Path() string {
+	return "/v2/keys.updateCredits"
+}
+
+func (h *Handler) Handle(ctx context.Context, s *zen.Session) error {
+	h.Logger.Debug("handling request", "requestId", s.RequestID(), "path", "/v2/keys.updateCredits")
+
+	// Authentication
+	auth, err := h.Keys.VerifyRootKey(ctx, s)
+	if err != nil {
+		return err
+	}
+
+	// Request validation
+	req, err := zen.BindBody[Request](s)
+	if err != nil {
+		return err
+	}
+
+	key, err := db.Query.FindKeyByIdOrHash(ctx,
+		h.DB.RO(),
+		db.FindKeyByIdOrHashParams{
+			ID:   sql.NullString{String: req.KeyId, Valid: true},
+			Hash: sql.NullString{String: "", Valid: false},
+		},
+	)
+	if err != nil {
+		if db.IsNotFound(err) {
+			return fault.Wrap(
+				err,
+				fault.Code(codes.Data.Key.NotFound.URN()),
+				fault.Internal("key does not exist"),
+				fault.Public("We could not find the requested key."),
+			)
+		}
+
+		return fault.Wrap(err,
+			fault.Code(codes.App.Internal.ServiceUnavailable.URN()),
+			fault.Internal("database error"),
+			fault.Public("Failed to retrieve Key information."),
+		)
+	}
+
+	// Validate key belongs to authorized workspace
+	if key.WorkspaceID != auth.AuthorizedWorkspaceID {
+		return fault.New("key not found",
+			fault.Code(codes.Data.Key.NotFound.URN()),
+			fault.Internal("key belongs to different workspace"),
+			fault.Public("The specified key was not found."),
+		)
+	}
+
+	// Check if API is deleted
+	if key.Api.DeletedAtM.Valid {
+		return fault.New("key not found",
+			fault.Code(codes.Data.Key.NotFound.URN()),
+			fault.Internal("key belongs to deleted api"),
+			fault.Public("The specified key was not found."),
+		)
+	}
+
+	// Permission check
+	err = h.Permissions.Check(
+		ctx,
+		auth.KeyID,
+		rbac.Or(
+			rbac.T(rbac.Tuple{
+				ResourceType: rbac.Api,
+				ResourceID:   "*",
+				Action:       rbac.UpdateKey,
+			}),
+			rbac.T(rbac.Tuple{
+				ResourceType: rbac.Api,
+				ResourceID:   key.Api.ID,
+				Action:       rbac.UpdateKey,
+			}),
+		),
+	)
+	if err != nil {
+		return err
+	}
+
+	if (req.Operation == openapi.Decrement || req.Operation == openapi.Increment) && (!req.Value.IsSpecified() || req.Value.IsNull()) {
+		return fault.New("wrong operation usage",
+			fault.Code(codes.App.Validation.InvalidInput.URN()),
+			fault.Internal("wrong operation usage"),
```

Comment:
I probably lead with a bad example here, but we don't need to use fault.Internal when creating new errors via fault.New.

the internal error logging will use the first argument to fault.New to construct an error message.

not a blocker here, this is fine. just fyi

Reply:
i'll remove it when going over all the endpoints again at the end